### DISCUSSION TITLE
Disscussion about "Writing Better Code ?"

### DISCUSSION THREAD

**Discussion by MostPow3rful** [Cyfrin/foundry-full-course-cu#2746] [Q&A]
GM @PatrickAlphaC 
---
![image](https://github.com/user-attachments/assets/db112e25-047d-4473-b203-6a5ce61ce3d8)
---
the type of _amount is `uint256` it means it starts from `type(uint256).min == 0` to `type(uint256).max (very big number)` so the variable with `uint256` type can not be less than `0` yes? so why we wrote `_amount <= 0` and we don't wrote `_amount == 0` and then reverting an error ?

**Comment by EngrPips** [comment]
This is very logical, and you are right. but i mean doing that way still do the check properly.
